extensions/omni_var/omni_var.c

```diff
@@ -0,0 +1,231 @@
+/**
+ * @file omni_var.c
+ *
+ */
+
+// clang-format off
+#include <postgres.h>
+#include <fmgr.h>
+// clang-format on
+#include <access/xact.h>
+#include <utils/builtins.h>
+#include <utils/guc.h>
+#include <utils/hsearch.h>
+#include <utils/lsyscache.h>
+#include <utils/memutils.h>
+
+PG_MODULE_MAGIC;
+
+PG_FUNCTION_INFO_V1(set);
+
+PG_FUNCTION_INFO_V1(get);
+
+static TransactionId last_used_txnid = InvalidTransactionId;
+static HTAB *current_tab = NULL;
+
+typedef struct VariableValue {
+  bool isnull;
+  Oid oid;
+  SubTransactionId subxactid;
+  Datum value;
+  struct VariableValue *previous;
+} VariableValue;
+
+typedef struct {
+  NameData name;
+  VariableValue variable_value;
+} Variable;
+
+static int nvars = 0;
+
+void _PG_init() {
+  DefineCustomIntVariable(
+      "omni_var.estimated_initial_txn_variables_count", "Estimated number of transaction variables",
+      "Estimated number of transaction variables, changes apply to the next transaction", &nvars,
+      1024, 0, 65535, PGC_USERSET, 0, NULL, NULL, NULL);
```

Comment:
(Given transactionId is 32 bit integer) Is 65535 (2^16) the maximum number of variables that one can define within a transaction?

Reply:
This is not a limit, but estimation and transaction IDs have nothing to do with the estimation or limit.